File: android/src/com/boliao/eod/WeatherWorkerThread.java
package com.boliao.eod;

import android.os.Handler;
import android.os.HandlerThread;

public class WeatherWorkerThread extends HandlerThread {
    private Handler handler;

    public WeatherWorkerThread() {
        super("WeatherWorkerThread");
    }

    public void prepareHandler() {
       handler = new Handler(getLooper());
    }

    public void postTask(Runnable r) {
        handler.post(r);
    }

    public void postTaskDelayed(Runnable r, long delayMillis) {
        handler.postDelayed(r, delayMillis);
    }
}

File: android/src/com/boliao/eod/WeatherWorkerThread.kt
package com.boliao.eod

import android.os.Handler
import android.os.HandlerThread

/**
 * A HandlerThread for weather runnables (jobs) to land on.
 * - has a Looper that constantly conveys Runnables to be acted on
 * - has a Handler to post Runnables onto the Looper
 *   Handler can postDelayed Runnables
 */
class WeatherWorkerThread : HandlerThread("WeatherWorkerThread") {
    private lateinit var handler: Handler

    // need this as looper only available after initializing HandlerThread
    fun prepareHandler() {
        handler = Handler(looper)
    }

    fun postTask(r: Runnable) {
        handler.post(r)
    }

    fun postTaskDelayed(r: Runnable, delayMillis: Long) {
        handler.postDelayed(r, delayMillis)
    }
}